### PR TITLE
more customizations for demo 3

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
# What this PR does

This PR:
- remove ajax cache to allow caching all urls, especially API
- allow manage and reorder of metadata pane accordion. Any suggestions about var naming, dict

# Testing this PR

### CODE REVIEW

File: components/tools/OmeroWeb/omeroweb/settings.py [hunk 1]
```diff
@@ -653,6 +653,19 @@ def leave_none_unset_int(s):
           " ] or [\"Repository\", {\"viewname\": \"webindex\", "
           "\"query_string\": {\"experimenter\": -1}}, "
           "{\"title\": \"Repo\"}]'``")],
+    "omero.web.ui.metadata_pane":
+        ["METADATA_PANE",
```
Comment (atarkowska) [inline components/tools/OmeroWeb/omeroweb/settings.py:657]: this allows fully customize accordion unloading labels

Comment (atarkowska) [reply components/tools/OmeroWeb/omeroweb/settings.py:657]: @will-moore @manics @pwalczysko @joshmoore could you have a look if you are ok with that?

File: components/tools/OmeroWeb/omeroweb/settings.py [hunk 2]
```diff
@@ -653,6 +653,19 @@ def leave_none_unset_int(s):
...
+          '{"name": "tag", "label": "Tags", "index": "0"},'
+          '{"name": "map", "label": "Key-Value Pairs", "index": "1"},'
+          '{"name": "table", "label": "Tables", "index": "2"},'
+          '{"name": "file", "label": "Attachements", "index": "3"},'
+          '{"name": "comment", "label": "Comments", "index": "4"},'
+          '{"name": "rating", "label": "Ratings", "index": "5"},'
+          '{"name": "other", "label": "Others", "index": "6"}'
+          ']'),
+         json.loads,
+         ("Manage Metadata pane accrodion.")],
```
Comment (will-moore) [inline components/tools/OmeroWeb/omeroweb/settings.py]: typo 'accordion'

File: components/tools/OmeroWeb/omeroweb/settings.py (hunk 1 above)
Comment (will-moore) [reply components/tools/OmeroWeb/omeroweb/settings.py:657]: This seems like a nice idea. Is this planned to come into develop for 5.3.0?

Comment (atarkowska) [reply components/tools/OmeroWeb/omeroweb/settings.py:657]: yes if we are happy with that config, this PR can be rebased against main line

File: components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/annotations/metadata_general.html [hunk 3]
```diff
@@ -719,149 +718,20 @@ <h1 class="can-collapse defclose" data-name="details">
         
     <!-- ANNOTATIONS -->
 
+        <!-- templates are loaded based on config -->
+        {% for t in ome.metadata_pane|dictsort:"index" %}
```
Comment (will-moore) [inline components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/annotations/metadata_general.html:722]: Why do you need an "index" on each item in the list for sorting? The list is already in the correct order, isn't it?

Comment (atarkowska) [reply components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/annotations/metadata_general.html:722]: this is true if you set, but for append it is easier to have that extra var, e.g. lets say you want to move `'{"name": "table", "label": "Tables", "index": 2},'` for the very end. You only have to:

```
bin/omero config remove omero.web.ui.metadata_pane '{"name": "table", "label": "Tables", "index": 2},'
bin/omero config append omero.web.ui.metadata_pane '{"name": "table", "label": "Tables", "index": 7},'
```

we do not have `bin/omero append index`

Comment (atarkowska) [reply components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/annotations/metadata_general.html:722]: I was also thinking if index shouldn't be 10, 20, 30,... that will help to shovel them in between easier

Comment (will-moore) [reply components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/annotations/metadata_general.html:722]: Right, if you don't have indexes them spaced out, 10, 20, 30, etc, then you can't insert one.
I guess it's just a decision as to what's the nicest default indexes to use. E.g can you insert `"1.1"` into `["1", "2"]`?

Comment (atarkowska) [reply components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/annotations/metadata_general.html:722]: sorry they should be numbers not strings :) then user can add 1, 1.1, 2

File: components/tools/OmeroWeb/omeroweb/webclient/views.py [hunk 4]
```diff
@@ -372,9 +372,8 @@ def logout(request, conn=None, **kwargs):
 
 
 ###########################################################################
-@login_required()
-@render_response()
-def load_template(request, menu, conn=None, url=None, **kwargs):
+def _load_template(request, menu, conn=None, url=None, **kwargs):
```
Comment (atarkowska) [inline components/tools/OmeroWeb/omeroweb/webclient/views.py:375]: @will-moore I made a clean up in load_template and now allow to override it. Could you please have a look if that won't break anything?

File: components/tools/OmeroWeb/omeroweb/settings.py [hunk 5]
```diff
@@ -653,6 +653,19 @@ def leave_none_unset_int(s):
           " ] or [\"Repository\", {\"viewname\": \"webindex\", "
           "\"query_string\": {\"experimenter\": -1}}, "
           "{\"title\": \"Repo\"}]'``")],
+    "omero.web.ui.metadata_pane":
```
Comment (will-moore) [inline components/tools/OmeroWeb/omeroweb/settings.py:656]: This should probably be a plural since it's a list. E.g. `metadata_panes or metadata_panels`?

**Comment by will-moore** [comment]
Seems to be working fine (testing locally).
You might want to start with 1 as the first index instead of 0, so you're not forced to use negative numbers to add at the first slot (although this works OK):

```
$ omero config append omero.web.ui.metadata_pane '{"name": "comment", "label": "Things people said", "index": -0.9}'
```

screen shot 2016-10-07 at 14 00 29

**Comment by joshmoore** [comment]
Do we no longer need https://github.com/openmicroscopy/openmicroscopy/pull/3896/files#diff-32b25f7075b9602021940173873c7224R119 then?

**Comment by atarkowska** [comment]
I wasn't sure how to backport that as it conflicts. I only handled webclient part.

**Comment by joshmoore** [comment]
Ok. Added a note to https://trello.com/c/1h5YDICh/205-populate-metadata

File: components/tools/OmeroWeb/omeroweb/webclient/tree.py [hunk 6]
```diff
@@ -1787,7 +1787,7 @@ def marshal_annotations(conn, project_ids=None, dataset_ids=None,
             join fetch ch.details.creationEvent
             join fetch ch.details.owner
             left outer join fetch ch.file as file
-            where %s
+            where %s order by ch.ns
```
Comment (atarkowska) [inline components/tools/OmeroWeb/omeroweb/webclient/tree.py:1790]: order of maps is hardcoded by namespace, I found that the most suitable and guarantee the same order across all the studies. Not sure the best solution

**Comment by joshmoore** [comment]
Merging this for the IDR-0.3.0-rc1 tag. Testing took place on the demo3 servers. Thanks all!

**Comment by atarkowska** [comment]
@jburel @will-moore  in this PR I added `omero.web.ui.metadata_pane` which probably is not the best choice for rebase PR to go to the main line. Does it make sense to you if in a main line we'll create `omero.client.ui.metadata.accordion.labels` or similar ? That would need to be applied in insight as well?

**Comment by will-moore** [comment]
The `omero.web.ui.metadata_pane` config is nice in that it allows you to remove panes you don't want. However, it's probably not very easy to add new panes from external apps here.
So, I'm not sure we could document it as a way to add content - Just to rename or remove panes.
Don't know how easy it is to implement the same in Insight? If it's doable, then we could consider supporting that in both clients. If not, then we could just go for the 'labels' only option.

**Comment by joshmoore** [comment]
@aleksandra-tarkowska : do you have a follow-up card for these items?

**Comment by atarkowska** [comment]
it is not really a follow up, but the place to report to https://trello.com/c/Ds2XzT9J/8-demo3-mapr-cache-web-api. For a bigger discussion or RFE put new card in the same column

**Pull request merged**